pkg/schedule/splitter/region_splitter.go

```diff
@@ -56,15 +56,17 @@ func NewSplitRegionsHandler(cluster sche.ClusterInformer, oc *operator.Controlle
 
 // RegionSplitter handles split regions
 type RegionSplitter struct {
-	cluster sche.ClusterInformer
-	handler SplitRegionsHandler
+	cluster           sche.ClusterInformer
+	handler           SplitRegionsHandler
+	addSuspectRegions func(ids ...uint64)
 }
 
 // NewRegionSplitter return a region splitter
-func NewRegionSplitter(cluster sche.ClusterInformer, handler SplitRegionsHandler) *RegionSplitter {
+func NewRegionSplitter(cluster sche.ClusterInformer, handler SplitRegionsHandler, addSuspectRegions func(ids ...uint64)) *RegionSplitter {
```

Comment:
Do we have other ways to avoid pass a function to it?

Reply:
Any better idea?